test/Microsoft.AspNet.Mvc.Core.Test/DefaultControllerTypeProviderTest.cs

```diff
@@ -211,7 +274,97 @@ public class NoSuffix : Mvc.Controller
     {
     }
 
+    public class NoSuffixPoco
+    {
+
+    }
+
     public class PocoController
     {
     }
+
+    public class CustomBaseController
+    {
+
+    }
+
+    public abstract class CustomAbstractBaseController
+    {
+
+    }
+
+    public class ChildWithoutSuffix : CustomBaseController
+    {
+
+    }
+
+    public class DescendantLevel1 : CustomBaseController
+    {
+
+    }
+
+    public class DescendantLevel2 : DescendantLevel1
+    {
+
+    }
+
+    public class AbstractChildWithoutSuffix : CustomAbstractBaseController
+    {
+
+    }
+
+    [NonController]
+    public class BasePocoNonControllerController
+    {
+
+    }
+
+    public class BasePocoNonControllerControllerChild : BasePocoNonControllerController
+    {
+
+    }
+
+    [NonController]
+    public class BaseNonControllerController : Controller
+    {
+
+    }
+
+    public class BaseNonControllerControllerChild : BaseNonControllerController
+    {
+
+    }
+
+    [NonController]
+    public class PocoNonController
```

Comment:
This is identical to `BasePocoNonControllerController` (not tested) I assume its just left over.